dbms/src/Debug/dbgFuncCoprocessorUtils.cpp

```diff
@@ -0,0 +1,166 @@
+            }
+            break;
+        }
+        else
+        {
+            result_data.emplace_back(std::move(block));
+        }
+    }
+
+    if (result_data.size() > 1)
+        throw Exception("Result block should be less than 128M!", ErrorCodes::BAD_ARGUMENTS);
+    return result_data[0];
+}
+
+bool dagRspEqual(Context & context, const tipb::SelectResponse & expected, const tipb::SelectResponse & actual, String & unequal_msg)
+{
+    auto schema = getSelectSchema(context);
+    SortDescription sort_desc = generateSDFromSchema(schema);
+    Block block_a = getMergedBigBlockFromDagRsp(context, schema, expected);
+    sortBlock(block_a, sort_desc);
+    Block block_b = getMergedBigBlockFromDagRsp(context, schema, actual);
+    sortBlock(block_b, sort_desc);
+    bool equal = blockEqual(block_a, block_b, unequal_msg);
+    if (!equal)
+    {
+        unequal_msg = fmt::format("{}\nExpected Results: \n{}\nActual Results: \n{}", unequal_msg, formatBlockData(block_a), formatBlockData(block_b));
+    }
+    return equal;
+}
+
+static const String ENCODE_TYPE_NAME = "encode_type";
+static const String TZ_OFFSET_NAME = "tz_offset";
+static const String TZ_NAME_NAME = "tz_name";
+static const String COLLATOR_NAME = "collator";
+static const String MPP_QUERY = "mpp_query";
+static const String USE_BROADCAST_JOIN = "use_broadcast_join";
+static const String MPP_PARTITION_NUM = "mpp_partition_num";
+static const String MPP_TIMEOUT = "mpp_timeout";
+
+DAGProperties getDAGProperties(const String & prop_string)
+{
+    DAGProperties ret;
+    if (prop_string.empty())
+        return ret;
+    std::unordered_map<String, String> properties;
+    Poco::StringTokenizer string_tokens(prop_string, ",");
+    for (const auto & string_token : string_tokens)
+    {
+        Poco::StringTokenizer tokens(string_token, ":");
+        if (tokens.count() != 2)
+            continue;
+        properties[Poco::toLower(tokens[0])] = tokens[1];
+    }
+
+    if (properties.find(ENCODE_TYPE_NAME) != properties.end())
+        ret.encode_type = properties[ENCODE_TYPE_NAME];
+    if (properties.find(TZ_OFFSET_NAME) != properties.end())
+        ret.tz_offset = std::stol(properties[TZ_OFFSET_NAME]);
+    if (properties.find(TZ_NAME_NAME) != properties.end())
+        ret.tz_name = properties[TZ_NAME_NAME];
+    if (properties.find(COLLATOR_NAME) != properties.end())
+        ret.collator = std::stoi(properties[COLLATOR_NAME]);
+    if (properties.find(MPP_QUERY) != properties.end())
+        ret.is_mpp_query = properties[MPP_QUERY] == "true";
+    if (properties.find(USE_BROADCAST_JOIN) != properties.end())
+        ret.use_broadcast_join = properties[USE_BROADCAST_JOIN] == "true";
+    if (properties.find(MPP_PARTITION_NUM) != properties.end())
+        ret.mpp_partition_num = std::stoi(properties[MPP_PARTITION_NUM]);
+    if (properties.find(MPP_TIMEOUT) != properties.end())
+        ret.mpp_timeout = std::stoi(properties[MPP_TIMEOUT]);
```

Comment:
```suggestion
    Poco::StringTokenizer string_tokens(prop_string, ",");
    for (const auto & string_token : string_tokens)
    {
        Poco::StringTokenizer tokens(string_token, ":");
        if (tokens.count() != 2)
            continue;
        if (Poco::toLower(tokens[0]) == ENCODE_TYPE_NAME)
            ret.encode_type = tokens[1];
        else if (Poco::toLower(tokens[0]) == TZ_OFFSET_NAME)
            ret.tz_offset = tokens[1];
        else if (properties.find(TZ_NAME_NAME) != properties.end())
        ...
    }
```
Does it looks better?

Reply:
I will look deep into the code and do more refinements of dbgFuncCoprocessor in this pr.